gists/helm-component-extraction/argocd/UPGRADE.md

```diff
@@ -0,0 +1,39 @@
+# Upgrade ArgoCD
+
+## From 2.1.7 to 2.5.6
+
+### Back up existing applications
+
+```shell
+make applications-snapshot
+```
+
+### Upgrade ArgoCD
+
+```shell
+# Prepare upgraded values
+make configure-helm-chart
+# Install upgraded chart with values
+make install-helm-chart
```

Comment:
make install-helm-chart
kubectl apply -f build/namespace.yaml
Warning: resource namespaces/argocd is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
namespace/argocd configured
kubectl apply -f build/argocd-secret.yaml
Warning: resource externalsecrets/argocd-secret is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
externalsecret.kubernetes-client.io/argocd-secret configured
helm repo add argo https://argoproj.github.io/argo-helm
"argo" already exists with the same configuration, skipping
helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "autoscaler" chart repository
...Successfully got an update from the "eks" chart repository
...Successfully got an update from the "aws-ebs-csi-driver" chart repository
...Successfully got an update from the "external-secrets" chart repository
...Successfully got an update from the "grafana" chart repository
...Successfully got an update from the "argo" chart repository
...Successfully got an update from the "prometheus-community" chart repository
Update Complete. ⎈Happy Helming!⎈
helm upgrade --install --namespace argocd argocd argo/argo-cd --version 5.17.1 --values build/values.yaml 
Error: UPGRADE FAILED: chart requires kubeVersion: >=1.22.0-0 which is incompatible with Kubernetes v1.21.14-eks-fb459a0
make: *** [Makefile:134: install-helm-chart] Error 1

Reply:
Interesting. Nice find